lib/src/model/getter_setter_combo.dart

```diff
@@ -110,8 +110,37 @@ mixin GetterSetterCombo on ModelElement {
   String get constantValueTruncated =>
       linkifyConstantValue(truncateString(constantValueBase, 200));
 
-  late final String constantValueBase = const HtmlEscape(HtmlEscapeMode.unknown)
-      .convert(constantInitializer?.toString() ?? '');
+  late final String constantValueBase = () {
+    final constantInitializer = this.constantInitializer;
+    if (constantInitializer == null) {
+      return '';
+    }
+
+    var initializerString = constantInitializer.toString();
+
+    final self = this;
+    if (self is! EnumField ||
+        constantInitializer is! InstanceCreationExpression) {
+      return _htmlEscape.convert(initializerString);
+    }
+
+    initializerString = 'const $initializerString';
+
+    var isImplicitConstructorCall = constantInitializer
+            .constructorName.staticElement?.isDefaultConstructor ??
+        false;
+    if (isImplicitConstructorCall) {
+      // For an enum value with an implicit constructor call (like
+      // `enum E { one, two; }`), `constantInitializer.toString()` does not
+      // include the implicit enum index argument (it is something like
+      // `const E()`). We must manually include it.
```

Comment:
Perhaps link the issue here and revisit this if it's fixed in the future.

Reply:
Done.